Analisis de la Felicidad Phyton deber

In [2]:
import csv
import pandas as pd

In [4]:
felicidad = pd.read_csv("felicidad.csv")
felicidad.head(10)

pais  anio  escalera_vida   log_pib  soporte_social  \
0  Afghanistán  2008       3.723590  7.168690        0.450662   
1  Afghanistán  2009       4.401778  7.333790        0.552308   
2  Afghanistán  2010       4.758381  7.386629        0.539075   
3  Afghanistán  2011       3.831719  7.415019        0.521104   
4  Afghanistán  2012       3.782938  7.517126        0.520637   
5  Afghanistán  2013       3.572100  7.522238        0.483552   
6  Afghanistán  2014       3.130896  7.516955        0.525568   
7  Afghanistán  2015       3.982855  7.500539        0.528597   
8  Afghanistán  2016       4.220169  7.497038        0.559072   
9  Afghanistán  2017       2.661718  7.497755        0.490880   

   expectativa_vida  libertad  generosidad  percepcion_corrupcion  \
0         50.799999  0.718114     0.177889               0.881686   
1         51.200001  0.678896     0.200178               0.850035   
2         51.599998  0.600127     0.134353               0.706766   
3         51.919998  0.495901     0.172137               0.731109   
4         52.240002  0.530935     0.244273               0.775620   
5         52.560001  0.577955     0.070403               0.823204   
6         52.880001  0.508514     0.113184               0.871242   
7         53.200001  0.388928     0.089091               0.880638   
8         53.000000  0.522566     0.051365               0.793246   
9         52.799999  0.427011    -0.112198               0.954393   

   afecto_positivo  afecto_negativo  confianza  calidad_democracia  \
0         0.517637         0.258195   0.612072           -1.929690   
1         0.583926         0.237092   0.611545           -2.044093   
2         0.618265         0.275324   0.299357           -1.991810   
3         0.611387         0.267175   0.307386           -1.919018   
4         0.710385         0.267919   0.435440           -1.842996   
5         0.620585         0.273328   0.482847           -1.879709   
6         0.531691         0.374861   0.409048           -1.773257   
7         0.553553         0.339276   0.260557           -1.844364   
8         0.564953         0.348332   0.324990           -1.855426   
9         0.496349         0.371326   0.261179           -1.886566   

   calidad_entrega  de_escalera_pais_anio  gini_banco_mundial  \
0        -1.655084               1.774662                 NaN   
1        -1.635025               1.722688                 NaN   
2        -1.617176               1.878622                 NaN   
3        -1.616221               1.785360                 NaN   
4        -1.404078               1.798283                 NaN   
5        -1.403036               1.223690                 NaN   
6        -1.312503               1.395396                 NaN   
7        -1.291594               2.160618                 NaN   
8        -1.392713               1.796219                 NaN   
9        -1.437808               1.454051                 NaN   

   gini_banco_mundial_promedio  
0                          NaN  
1                          NaN  
2                          NaN  
3                          NaN  
4                          NaN  
5                          NaN  
6                          NaN  
7                          NaN  
8                          NaN  
9                          NaN

Calculo del promedio y desviacion estandar de la felicidad para el año 2005

In [17]:
pAnio = felicidad[felicidad['anio'] == 2005]
pAnio_avg = pAnio['escalera_vida'].mean()
print('El promedio de felicidad para el año 2005 es:',pAnio_avg)

pdvAnio = felicidad[felicidad['anio'] == 2005]
panio_dv = pdvAnio['escalera_vida'].std()
print('La desviacion estandar de felicidad para el año 2005 es:',panio_dv)

El promedio de felicidad para el año 2005 es: 6.446164272449635
La desviacion estandar de felicidad para el año 2005 es: 0.9191426322726484


Calculo del promedio y desviacion estandar de la felicidad para el anio 2018

In [18]:
uAnio = felicidad[felicidad['anio'] == 2018]
uAnio_avg = uAnio['escalera_vida'].mean()
print('El promedio de felicidad para el año 2018 es:',uAnio_avg)

udvAnio = felicidad[felicidad['anio'] == 2018]
uanio_dv = udvAnio['escalera_vida'].std()
print('La desviacion estandar de felicidad para el año 2018 es:',uanio_dv)

El promedio de felicidad para el año 2018 es: 5.502134340650895
La desviacion estandar de felicidad para el año 2018 es: 1.1034612436939357


Indice de felicidad de Alemania años 2005 y 2018

In [19]:
Alemania1 = felicidad[(felicidad['pais'] == 'Alemania') & (felicidad['anio'] == 2005)]
felicidadAlemania1 = Alemania1['escalera_vida'].values
print('La felicidad en Alemania para el 2005 fue:', felicidadAlemania1)

Alemania2 = felicidad[(felicidad['pais'] == 'Alemania') & (felicidad['anio'] == 2018)]
felicidadAlemania2 = Alemania2['escalera_vida'].values
print('La felicidad en Alemania para el 2018 fue:', felicidadAlemania2)

La felicidad en Alemania para el 2005 fue: [6.61954975]
La felicidad en Alemania para el 2018 fue: [7.11836433]


Indice de felicidad de Bolivia años 2006 y 2018

In [20]:
Bolivia1 = felicidad[(felicidad['pais'] == 'Bolivia') & (felicidad['anio'] == 2006)]
felicidadBolivia1 = Bolivia1['escalera_vida'].values
print('La felicidad en Bolivia para el 2006 fue:', felicidadBolivia1)

Bolivia2 = felicidad[(felicidad['pais'] == 'Bolivia') & (felicidad['anio'] == 2018)]
felicidadBolivia2 = Bolivia2['escalera_vida'].values
print('La felicidad en Bolivia para el 2018 fue:', felicidadBolivia2)

La felicidad en Bolivia para el 2006 fue: [5.37398624]
La felicidad en Bolivia para el 2018 fue: [5.91573429]


Comparacion del indice de generosidad de Alemania y Bolivia vs resto del mundo

In [21]:
promedio_generosidad = felicidad['generosidad'].mean()
print("Promedio de generosidad mundial:", promedio_generosidad)

Promedio de generosidad mundial: 7.86347701148664e-05


In [22]:
Alemania3 = felicidad[(felicidad['pais'] == 'Alemania')]
generosidad_Alemania= Alemania3['generosidad'].mean()
print('El promedio de generosidad en Alemania es:', generosidad_Alemania)

El promedio de generosidad en Alemania es: 0.09694470685314047


In [23]:
Bolivia3 = felicidad[(felicidad['pais'] == 'Bolivia')]
generosidad_Bolivia= Bolivia3['generosidad'].mean()
print('El promedio de generosidad en Bolivia es:', generosidad_Bolivia)

El promedio de generosidad en Bolivia es: -0.04303260491444511


Este analisis nos resume que para Alemania en 2005 y 2018 presento un indice de felicidad de 6.61 y 7.11 respectivamente, mientras que para Bolivia mostro para 2006 un indicador de 5.37 y 2018 5.91, por lo tanto Alemania fue mas feliz que Bolivia en segun estos años particulares.
En promedio la felicidad mundial es 5.5 por lo tanto Alemania al mostrar un indice de felicidad mas alto se ubica mejor en este ranking, mientras que Bolivia esta cerca del promedio.
Hemos utilizado como punto de referencia el análisis de la columna que evalúa la generosidad, donde se ha calculado un promedio para Alemania y Bolivia en todos los años respectivamente, y se ha comparado con el promedio a nivel mundial. El promedio global de generosidad se sitúa en aproximadamente 0.00003326. A continuación, Bolivia se encuentra con un valor de -0.043 y Alemania con 0.09, lo que indica que las personas alemanas son más generosas que las bolivianas.